### PR TITLE
Rewrite custom form check backgrounds

### LINKED ISSUE #24598: Custom Radio and Checkbox broken when $enable-gradients is true
unbenannt-2

### PR DESCRIPTION
**Pull request by mdo**
Previously, this was all just a `background-color` and `background-image`. When we restored the gradients though, we had two `background-image`s competing on the same element, causing rendering glitches. This refactors that code, creating a mixin to simplify things, so we can we easily use two `background-image`s (SVG icon and gradient) when `$enable-gradients` is set to true.

Fixes #24598.

---

#### Todo
- [x] Remove variable changes
- [x] Double check validation states of custom checks

### CODE REVIEW

**Comment by jonnysp** [comment]
@mdo thanks looks good. maybe change the validation status?

File: scss/_variables.scss [hunk 1]
```diff
@@ -98,8 +98,8 @@ $yiq-text-light: $white !default;
 
 $enable-caret:              true !default;
 $enable-rounded:            true !default;
-$enable-shadows:            false !default;
-$enable-gradients:          false !default;
+$enable-shadows:            true !default;
```
Comment (andresgalante) [inline scss/_variables.scss]: Do we want to leave shadows and gradients on by default?

Comment (mdo) [reply scss/_variables.scss]: Nope! Not yet at least :).

**Comment by mdo** [comment]
Okay, decently sized update here:

I've had to change how we apply the `background-color` and multiple `background-image`s for custom checkboxes and radios. Previously, we had multiple comma-separated `background-image`s on the same `.custom-control-indicator` element. This presents a problem though for when you want to customize the background color or gradient without having to redeclare the icon, too. Now, the gradient and color are on `.custom-control-indicator` and the icon is on `.custom-control-indicator::before`.

I'll be adding a note to the migration docs for this here as well.

**Comment by mdo** [comment]
@jonnysp With the changes I mentioned in my previous comment, you should be good to go with green and red background gradients on those custom forms.

**Comment by jonnysp** [comment]
looks great, many thanks. can merge!

**Review by jonnysp — APPROVED** [review]
looks great, many thanks. can merge!

**Pull request merged**